src/pysdmx/io/xml/__structure_aux_reader.py

```diff
@@ -551,8 +551,8 @@ def __format_con_id(self, concept_ref: Dict[str, Any]) -> Dict[str, Any]:
         return rep
 
     @staticmethod
-    def __format_relationship(json_rel: Dict[str, Any]) -> Optional[str]:
-        att_level = None
+    def __format_relationship(json_rel: Dict[str, Any]) -> str:
+        att_level = "D"
 
         for scheme in [DIM, PRIM_MEASURE, MEASURE, DFW, OBSERVATION]:
```

Comment:
What is meant with `scheme` here? It reminds of SDMX item scheme, but I guess it's something else, as this would not be relevant to derive the attachment level?

Reply:
In the if/elif/else clause, I think we should change 2 things:

- We should check if the attachment level is set to group and, if so, we should throw an Unimplemented error (we don't yet support groups)
- The else clause should throw an Invalid error (any attachment level which is not one of the expected ones is probably invalid?)

Reply:
The names of the attachment levels have changed with the SDMX versions. Has this been taken into account?